sdk/servicebus/service-bus/src/core/autoLockRenewer.ts

```diff
@@ -19,27 +19,28 @@ export type RenewableMessageProperties = Readonly<
   // updated when we renew the lock
   Pick<ServiceBusMessageImpl, "lockedUntilUtc">;
 
+type MinimalLink = Pick<LinkEntity<any>, "name" | "logPrefix" | "entityPath">;
+
 /**
  * Tracks locks for messages, renewing until a configurable duration.
  *
  * @internal
  * @ignore
  */
-export class AutoLockRenewer {
+export class LockRenewer {
   /**
-   * @property {Map<string, Function>} _messageRenewLockTimers Maintains a map of messages for which
-   * the lock is automatically renewed.
+   * @property _messageRenewLockTimers A map of link names to individual maps for each
+   * link that map a message ID to its auto-renewal timer.
```

Comment:
What was concluded on the uniqueness of the message-id?

Reply:
I'd like to revisit this so we can look into it more (since we've had it this way for forever).

So no conclusion yet and this was "safe enough" for now. I think we could conceivably change this before GA.

Reply:
This is safe for now because we are not changing anything and maintaining status quo with what we had before, but lets log an issue to check what happens to messages that were not created with message id

Reply:
Filed as an issue: #11682